### PR TITLE
Configuration model

### LINKED ISSUE #270: Create configuration model
## Feature
Add configuration model with mongoose

* app_name: e.g. 'MHacks X'
* start_date: e.g. Date for september 24
* end_date
* is_livepage_enabled
* is_applications_open

### PR DESCRIPTION
**Pull request by AndrewBoland**
Fixes #270 .

### PR Description
Added configuration model and routes

### What needs to be reviewed?
server/db/model/Configuration.js
server/routes/api/configuration.js

### Context?
There is some extra code in server/routes/api/configuration.js that I don't know what to do with

### CODE REVIEW

File: server/routes/api/configuration.js [hunk 1]
```diff
@@ -30,10 +52,96 @@ router.get('/', function(req, res) {
                     configuration
                 });
             });
+    }
+    */
+});
+
+// Handles post requests for /v1/configuration/control
+//This is on route /configuration/control so it is not easily accessible (it should be limited to one configuration at a time)
+router.post('/contril', authMiddleware('admin', 'api'), function(req, res) {
```
Comment (antoniomika) [inline server/routes/api/configuration.js]: Typo

File: server/routes/api/configuration.js [hunk 2]
```diff
@@ -1,13 +1,35 @@
...
+            console.error(err);
+            res.send({
+                status: false,
+                message: Responses.UNKNOWN_ERROR
+            });
+        });
 
+    //TODO I don't know what this did so I didn't remove it yet
+    /*
     if (req.get('Authorization')) {
```
Comment (antoniomika) [inline server/routes/api/configuration.js]: I'm gonna guess this was for getting the configuration with a user object. Let's keep them as separate endpoints

File: server/db/model/Configuration.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/* eslint-disable */
```
Comment (antoniomika) [inline server/db/model/Configuration.js]: Enable ESLint

Comment (antoniomika) [inline server/db/model/Configuration.js]: Enable ESLint

**Comment by AndrewBoland** [comment]
Yeah there will only be one document, but it's not really enforce

File: server/routes/api/configuration.js [hunk 4]
```diff
@@ -1,41 +1,120 @@
...
+            } else {
+                res.send({
+                    status: false,
+                    configuration: {}
+                });
+            }
+        })
+        .catch(err => {
+            console.error(err);
+            res.send({
```
Comment (connorkrupp) [inline server/routes/api/configuration.js]: `res.status(500.send(...`

**Pull request merged**